go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressNetworkFirewallRuleType indicates whether an EgressNetworkFirewallRule allows or denies traffic
+// +kubebuilder:validation:Pattern=^Allow|Deny$
+type EgressFirewallRuleType string
+
+const (
+	EgressFirewallRuleAllow EgressFirewallRuleType = "Allow"
+	EgressFirewallRuleDeny  EgressFirewallRuleType = "Deny"
+)
+
+// +genclient
+// +resource:path=egressfirewall
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="EgressFirewall Status",type=string,JSONPath=".status.egressFirewallStatus"
+// EgressFirewall describes the current egress firewall for a Namespace.
+// Traffic from a pod to an IP address outside the cluster will be checked against
+// each EgressFirewallRule in the pod's namespace's EgressFirewall, in
+// order. If no rule matches (or no EgressFirewall is present) then the traffic
+// will be allowed by default.
+type EgressFirewall struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of EgressFirewall.
+	Spec EgressFirewallSpec `json:"spec"`
+	// Observed status of EgressFirewall
+	// +optional
+	Status EgressFirewallStatus `json:"status,omitempty"`
+}
+
+// EgressFirewallStatus is the status of the EgressFirewall Object
+type EgressFirewallStatus struct {
+	// egressFirewallStatus contains the status of the EgressFirewall object
+	EgressFirewallObjectStatus string `json:"egressFirewallStatus"`
+}
+
+// EgressFirewallSpec is a desired state description of EgressFirewall.
+type EgressFirewallSpec struct {
+	// a collection of egress firewall rule objects
+	Rules []EgressFirewallRule `json:"egress"`
```

Comment:
why is the json field called `egress`, but the spec variable is called `Rules`?

Reply:
the egress comes from openshift, that was the openshift name

Reply:
The field name and the json name should match. It's too confusing otherwise.

Reply:
done